python/cudf/cudf/core/dataframe.py

```diff
@@ -1837,16 +1837,25 @@ def _make_operands_and_index_for_binop(
             rhs = {name: other for name in self._data}
             equal_columns = True
         elif isinstance(other, Series):
-            rhs = dict(zip(other.index.values_host, other.values_host))
+            rhs = dict(zip(other.index.to_pandas(), other.values_host))
             # For keys in right but not left, perform binops between NaN (not
             # NULL!) and the right value (result is NaN).
             left_default = as_column(np.nan, length=len(self))
-            equal_columns = other.index.to_pandas().equals(
-                self._data.to_pandas_index()
-            )
-            can_use_self_column_name = equal_columns or (
+            can_use_self_column_name = (
                 list(other._index._data.names) == self._data._level_names
             )
+            if cudf.get_option("mode.pandas_compatible"):
+                equal_columns = other.index.to_pandas().equals(
+                    self._data.to_pandas_index()
+                )
+                can_use_self_column_name = (
+                    equal_columns or can_use_self_column_name
+                )
+            else:
+                if not can_use_self_column_name:
+                    can_use_self_column_name = other.index.to_pandas().equals(
+                        self._data.to_pandas_index()
+                    )
```

Comment:
This should be logically equivalent:
```suggestion
            if not can_use_self_column_name:
                can_use_self_column_name = other.index.to_pandas().equals(
                    self._data.to_pandas_index()
                )
```

Reply:
Discussed offline and decided to remove pandas compatibility mode logic from here.